.github/actions/setup_build/action.yml

```diff
@@ -61,7 +84,7 @@ runs:
       uses: actions/cache/restore@v4
       with:
         path: llvm_install/**
-        key: llvm-${{ inputs.os }}-${{ inputs.os == 'ubuntu' && inputs.ubuntu_version || inputs.windows_version }}-${{ inputs.arch }}-v${{ inputs.llvm_version }}-${{ inputs.llvm_build_type }}
+        key: llvm-${{ inputs.os }}-${{ inputs.os == 'ubuntu' && inputs.ubuntu_version || inputs.windows_version }}-${{ inputs.cross_arch == '' && inputs.arch || inputs.cross_arch }}-v${{ inputs.llvm_version }}-${{ inputs.llvm_build_type }}
```

Comment:
can we have some comments here - or possibly via another action.

Reply:
I added another step to define the key sections concerned.